### ISSUE TITLE
SQLDialect.ASE cannot be used with a MockConnection

### ISSUE THREAD

**Issue by tobiasgirardet** [jOOQ/jOOQ#9468]
### Expected behavior and actual behavior:

It is currently not possible to use the SQLDialect.ASE with a MockConnection. Code fails with the following stacktrace:

```
        at org.jooq_3.12.1.ASE.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2717)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:755)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:383)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:501)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:470)
	at org.jooq.impl.SelectImpl.fetchLazy(SelectImpl.java:2751)

    ...
Caused by: java.sql.SQLFeatureNotSupportedException: Unsupported Operation
	at org.jooq.tools.jdbc.MockStatement.unwrap(MockStatement.java:844)
	at org.jooq.tools.jdbc.DefaultStatement.unwrap(DefaultStatement.java:146)
	at org.jooq.impl.Tools.isJtds(Tools.java:3958)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:3853)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:294)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:369)
	... 72 more
```

### Versions:

- jOOQ: 3.12.1
- Database (include vendor): MockConnection

**Comment by lukaseder** [comment]
Thank you very much for your report. This should be easy to fix. We'll include the fix for this week's planned 3.12.3 release.

**Comment by tobiasgirardet** [comment]
Thanks for the fast response! The following code snippet is quite self contained, but enough to trigger the same exception:

```
    @Test
    public void test() {
        MockConnection connection = new MockConnection(new MockDataProvider() {
            @Override
            public MockResult[] execute(MockExecuteContext mockExecuteContext) throws SQLException {
                return new MockResult[0];
            }
        });
        DSL.using(connection, SQLDialect.ASE).select().from().fetchLazy();
    }
```
